### ISSUE TITLE
parallel VUMPS uses the wrong environment functions

### ISSUE THREAD

**Issue by ebertok** [ITensor/ITensorInfiniteMPS.jl#35]
when using the parallel vumps method in vumps_localham.jl,

environments are calculated  in line 380 using 
```julia
 # Sum the Hamiltonian terms in the unit cell
  for n in 2:Nsites
    hᴸ[n] = hᴸ[n - 1] * ψ.AL[n] * ψ̃.AL[n] + hᴸ[n]
  end
  Hᴸ = left_environment(hᴸ, ψ; tol=krylov_tol)

  for n in 2:Nsites
    hᴿ[n] = hᴿ[n + 1] * ψ.AR[n + 1] * ψ̃.AR[n + 1] + hᴿ[n]
  end
  Hᴿ = right_environment(hᴿ, ψ; tol=krylov_tol)
```

,however the function `left_environment(hᴸ, ψ; tol=krylov_tol)` does not exist. 
I fixed it ad-hoc by copying the analogous code from the sequential algorithm, but I'm not sure if this is correct:

```julia
𝕙ᴸ = copy(hᴸ)
  for k in 1:Nsites
    𝕙ᴸ[k] = left_environment_cell(ψ, ψ̃, hᴸ, k)
  end
  Hᴸ = left_environment(hᴸ, 𝕙ᴸ, ψ; tol=krylov_tol)

  for n in 2:Nsites
    hᴿ[n] = hᴿ[n + 1] * ψ.AR[n + 1] * ψ̃.AR[n + 1] + hᴿ[n]
  end
  Hᴿ = right_environment(hᴿ, ψ; tol=krylov_tol)
```

**Comment by mtfishman** [comment]
Ah I see. It looks like a minor case of bit rot. I think I updated the code to fix an issue I was seeing with `left_environment` and `right_environment`, but didn't update the parallel version of the code.

In practice, I think I didn't bother to update the parallel version since I found the sequential version was converging better anyway, but maybe it is still worth trying to update the parallel version so we can test them out (if I remember correctly, one of them wasn't always better, it depends on the problem and unit cell size).

Please feel free to make a PR with the change you proposed above.

Out of curiosity, was there a particular reason you wanted to use the parallel version?

**Comment by ebertok** [comment]
Thank you for the swift reply.

I will make a new branch and do a pull request after some cleanup of my ad-hoc implementations.

My motivation was mostly for implementing time-evolution within the current vumps implementation. Somehow I have not managed to make it work using the sequential formulation, although also the parallel method has some quirks. (see also #36).

**Comment by mtfishman** [comment]
Closed by #37.
